### PR TITLE
feat: adding solidjs-vite framework

### PR DESCRIPTION
**Pull request by mzaien**
Issue:
maybe this [#18583](https://github.com/storybookjs/storybook/issues/18583) can be closed
## What I did
Implemented Vite builder coupled with Solid renderer using the official `vite-plugin-solid` plugin. 
## How to test

- [  ] Is this testable with Jest or Chromatic screenshots? 
- [ x ] Does this need a new example in the kitchen sink apps?
- [ x ] Does this need an update to the documentation?

not sure about if this is testable tbh, I followed steps made in this [pr](https://github.com/storybookjs/storybook/pull/20390) to give credit

### CODE REVIEW

File: code/frameworks/solidjs-vite/README.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+## Requirements
+
+- [Solid](https://www.solidjs.com/) >= 1.x
+- [Storybook](https://storybook.js.org/) >= 7.x
+
+## Getting Started
+
+### In a project without Storybook
+
+Follow the prompts after running this command in your Next.js project's root directory:
```
Comment (valentinpalkovic) [inline code/frameworks/solidjs-vite/README.md:12]:
```suggestion
Follow the prompts after running this command in your SolidJS project's root directory:
```

File: code/frameworks/solidjs-vite/README.md [hunk 2]
```diff
@@ -0,0 +1,46 @@
+# Storybook for Solid <!-- omit in toc -->
+
+## Requirements
+
+- [Solid](https://www.solidjs.com/) >= 1.x
+- [Storybook](https://storybook.js.org/) >= 7.x
+
+## Getting Started
+
+### In a project without Storybook
```
Comment (valentinpalkovic) [inline code/frameworks/solidjs-vite/README.md:10]: To automatically setup the solidjs-vite framework and to install it automatically for the user, you have to make some adjustments in https://github.com/storybookjs/storybook/blob/next/code/lib/cli/src/project_types.ts#L123 and https://github.com/storybookjs/storybook/blob/next/code/lib/cli/src/initiate.ts#L67

**Comment by valentinpalkovic** [comment]
@mzaien Thank you for your outstanding contribution! I left some comments. If you have questions, feel free to ask at any time.

File: code/frameworks/solidjs-vite/README.md (hunk 2 above)
Comment (mzaien) [reply code/frameworks/solidjs-vite/README.md:10]: @valentinpalkovic  I made changes to the needed files, if there is something else please guide me

File: code/lib/cli/src/project_types.ts [hunk 3]
```diff
@@ -81,6 +81,7 @@ export enum ProjectType {
   SVELTEKIT = 'SVELTEKIT',
   RAX = 'RAX',
   AURELIA = 'AURELIA',
+  SOLID = 'SOLID',
```
Comment (valentinpalkovic) [inline code/lib/cli/src/project_types.ts:84]: Also add `solid-vite` to the `SupportedFrameworks` union in line 19

File: code/lib/cli/src/generators/SOLID/index.ts [hunk 4]
```diff
@@ -0,0 +1,8 @@
+import { baseGenerator } from '../baseGenerator';
+import type { Generator } from '../types';
+
+const generator: Generator = async (packageManager, npmOptions, options) => {
+  await baseGenerator(packageManager, npmOptions, options, 'solid-js');
```
Comment (valentinpalkovic) [inline code/lib/cli/src/generators/SOLID/index.ts:5]:
```suggestion
  await baseGenerator(packageManager, npmOptions, options, 'html', undefined, 'solid-vite');
```

File: code/lib/cli/src/initiate.ts [hunk 5]
```diff
@@ -139,6 +140,11 @@ const installStorybook = (
           commandLog('Adding Storybook support to your "Angular" app\n')
         );
 
+      case ProjectType.SOLID:
+        return solidGenerator(packageManager, npmOptions, generatorOptions).then(
+          commandLog('Adding Storybook support to your "SOLID" app\n')
```
Comment (valentinpalkovic) [inline code/lib/cli/src/initiate.ts:145]:
```suggestion
          commandLog('Adding Storybook support to your "Solid" app\n')
```

File: code/frameworks/solidjs-vite/package.json [hunk 6]
```diff
@@ -0,0 +1,84 @@
+{
+  "name": "@storybook/solid-vite",
```
Comment (valentinpalkovic) [inline code/frameworks/solidjs-vite/package.json:2]: Go to `code/lib/cli/src/versions.ts` and add `@storybook/solid-vite` there as well

Comment (mzaien) [reply code/frameworks/solidjs-vite/package.json:2]: 👍

**Comment by valentinpalkovic** [comment]
@mzaien I want to give you a quick update: I will discuss your PR internally with the other Storybook maintainers. We'll discuss which criteria must be fulfilled to accept a new Storybook framework in the Storybook Monorepo. Due to the holiday season, I would like to ask you for some patience. I will let you know as soon as we have discussed your PR. Nevertheless, thank you for your incredible work!

**Comment by mzaien** [comment]
> @mzaien I want to give you a quick update: I will discuss your PR internally with the other Storybook maintainers. We'll discuss which criteria must be fulfilled to accept a new Storybook framework in the Storybook Monorepo. Due to the holiday season, I would like to ask you for some patience. I will let you know as soon as we have discussed your PR. Nevertheless, thank you for your incredible work!

Understood 👍, thanks for your time and guide

**Comment by mzaien** [comment]
Hello @ndelangen , 
not meaning to bother you do I need to do anything to move forward with this pr  😆 I would love to do anything to have it before storybook 7 release 🚀

**Comment by ndelangen** [comment]
@mzaien apologies for the delay, I've been dealing with some things on my side.

The storybook core team has decided that we need to focus on fixing bugs and get ready for the 7.0 release.

We therefore need to constraint a little what we can merge into the repo at this time.
We decided we can extend the storybook CLI to allow for new renderers/frameworks; but we won't be adding those renderers themselves to the repo at this time.

#20411 by @literalpie shows an example of this, the CLI makes external references to packages in community packages.

**Comment by literalpie** [comment]
@mzaien Have you started a standalone project for a solid storybook framework? If it's helpful, I can port some of this code to a new repo

**Comment by literalpie** [comment]
Nevermind, it looks like https://github.com/storybookjs/solidjs is already started

**Comment by mzaien** [comment]
> Nevermind, it looks like https://github.com/storybookjs/solidjs is already started

@literalpie oh did not know we can make new project under storybook, thanks for enlightening me

**Pull request closed without merging**